aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1880,8 +1880,13 @@ Tensor _matmul_impl(
 
 Tensor matmul(const Tensor & tensor1, const Tensor & tensor2) {
   auto maybe_outnames = namedinference::compute_matmul_outnames(tensor1, tensor2);
-  at::Tensor unused;
-  auto result = at::native::_matmul_impl(unused, tensor1, tensor2);
+  at::Tensor result, unused;
```

Comment:
Maybe add a note here describing why we need the is_nested() checks and possibly a TODO to remove them if we ever get per-dispatch-key backward functionality OR if we expand reshape in the future and rewrite this in terms of reshape + bmm?